### PR TITLE
E2E for snapshot feature

### PR DESCRIPTION
**Pull request by ambiknai**
1. Check addon version and run snapshot test [ along with basic volume tests ] if addon version is >= 5.0
2. New method `check_addon_pod_status` added to check controller pods are up or not once addon is enabled.  Keeping hardcoded values in sleep sometimes doesn't work as time taken for pod to come up vary.
3. Tested configuration and jenkins job build for existing and new cluster. 
4. As of now stage has known issue: https://jiracloud.swg.usma.ibm.com:8443/browse/SNAP-2293. Fix will be available on Monday as per discussion with VPC team.

### CODE REVIEW

File: tests/e2e/iks-cluster [hunk 1]
```diff
@@ -209,6 +208,27 @@ function createVPC {
...
 }
 
+function check_addon_pod_status {
+  set -x 
+  attempts=0
+  pod_name=$1
+  while true; do
+    attempts=$((attempts+1))
+    pod_status=$(kubectl get sts -n kube-system | grep "$pod_name"| awk '{print $2}')
+    if [   "$pod_status" = "1/1" ]; then
```
Comment (GunaKKIBM) [inline tests/e2e/iks-cluster:218]: Is the negative scenario tested, where pod is not running?

Comment (GunaKKIBM) [reply tests/e2e/iks-cluster:218]: Because I think == is needed instead of =

File: tests/e2e/iks-cluster [hunk 2]
```diff
@@ -1014,6 +1042,17 @@ else
 fi
 
 exportKubeConfig $cluster_name; rc=$?
+
+
```
Comment (GunaKKIBM) [inline tests/e2e/iks-cluster:1046]: Could you please add a comment here, that you are trying to revert to original addon here, if I am understanding correctly

File: tests/e2e/iks-cluster (hunk 1 above)
Comment (GunaKKIBM) [reply tests/e2e/iks-cluster:218]: Checked the results. This is working fine.

**Review by GunaKKIBM — APPROVED** [review]

File: tests/e2e/iks-cluster [hunk 3]
```diff
@@ -12,7 +12,7 @@
 E2E_TEST_SETUP="$EXECPATH/e2e-setup.out"
 
 default_IC_RES_GRP="vpc-e2e-test"
-default_IC_COS_INSTANCE="crn:v1:bluemix:public:cloud-object-storage:global:a/e242f140687cd68a8e037b26680e0f04:a15e0450-6de1-4c44-8ce0-804faceef026::"
+default_IC_COS_INSTANCE="crn:v1:staging:public:cloud-object-storage:global:a/8ee729d7f903db130b00257d91b6977f:a5ecf8b4-36b5-47e4-9f21-caf77ec2868e::"
```
[the remark this answers is not in the document]
Comment (ambiknai) [reply tests/e2e/iks-cluster:15]: @meghnasiingh  You are right. this one is for stage. Old value mentioned is no more valid. So got new key from Akash and updated the script.

[the remark this answers is not in the document]
Comment (meghnasiingh) [reply tests/e2e/iks-cluster:15]: ok

**Review by meghnasiingh — APPROVED** [review]

File: tests/e2e/iks-cluster [hunk 4]
```diff
@@ -209,6 +208,27 @@ function createVPC {
...
+      echo "$pod_name  not running well."
+      kubectl get pods -n kube-system| awk "/$pod_name-/"
+      exit 1
+    fi
+    echo "$pod_name state == $pod_status  Sleeping 10 seconds"
+    sleep 10
+  done
+}
+
 function createSubnet {
```
Comment (ambiknai) [inline tests/e2e/iks-cluster:232]: check for node pods as well.

Comment (ambiknai) [reply tests/e2e/iks-cluster:232]: sleep at 30s

File: tests/e2e/iks-cluster [hunk 5]
```diff
@@ -1014,6 +1042,18 @@ else
...
 exportKubeConfig $cluster_name; rc=$?
+
+# From Jenkins there is option to mention addon version. 
+# So if value is not "default", disable and enable user specified version
+if [[ $e2e_addon_version != "default"  ]]; then
+        echo "Info: Installing addon $ADDON_VERSION"
+        ibmcloud ks cluster addon disable vpc-block-csi-driver -f --cluster $cluster_name
+        sleep 60
+        ibmcloud ks cluster addon enable vpc-block-csi-driver --cluster $cluster_name --version $e2e_addon_version
+        sleep 10
```
Comment (ambiknai) [inline tests/e2e/iks-cluster:1053]: Sleep 30s

File: tests/e2e/iks-cluster [hunk 6]
```diff
@@ -209,6 +208,27 @@ function createVPC {
...
+    if [   "$pod_status" = "1/1" ]; then
+      echo "$pod_name is  running ."
+      break
+    fi
+    if [[ $attempts -gt 30 ]]; then
+      echo "$pod_name  not running well."
+      kubectl get pods -n kube-system| awk "/$pod_name-/"
+      exit 1
+    fi
+    echo "$pod_name state == $pod_status  Sleeping 10 seconds"
```
Comment (ambiknai) [inline tests/e2e/iks-cluster:227]: print time elapsedkkk

**Pull request merged**